x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformPersistentTasksExecutor.java

```diff
@@ -145,11 +145,19 @@ protected void nodeOperation(AllocatedPersistentTask task, @Nullable DataFrameTr
         // <5> load next checkpoint
         ActionListener<DataFrameTransformCheckpoint> getTransformNextCheckpointListener = ActionListener.wrap(
                 nextCheckpoint -> {
-                    indexerBuilder.setNextCheckpoint(nextCheckpoint);
+
+                    if (nextCheckpoint.isEmpty()) {
+                        // corner case which should not happen ;-)
+                        // reset the position to force a full re-run with checkpoint creation
```

Comment:
`getTransformCheckpoint` always returns a checkpoint object: in case it does not find a document in the internal index, it returns an empty object.

(the comment is misleading, I will re-phrase)